crates/api/src/site/registration_applications/approve.rs

```diff
@@ -44,15 +44,23 @@ impl Perform for ApproveRegistrationApplication {
       .accepted_application(Some(data.approve))
       .build();
 
-    let approved_user_id = registration_application.local_user_id;
-    LocalUser::update(context.pool(), approved_user_id, &local_user_form).await?;
+    let assessed_user_id = registration_application.local_user_id;
+    LocalUser::update(context.pool(), assessed_user_id, &local_user_form).await?;
 
     if data.approve {
-      let approved_local_user_view = LocalUserView::read(context.pool(), approved_user_id).await?;
-
+      let approved_local_user_view = LocalUserView::read(context.pool(), assessed_user_id).await?;
+    
       if approved_local_user_view.local_user.email.is_some() {
         send_application_approved_email(&approved_local_user_view, context.settings())?;
       }
+    } else {
+      let denied_local_user_view = LocalUserView::read(context.pool(), assessed_user_id).await?;
+    
+      if denied_local_user_view.local_user.email.is_some() {
+        send_application_denied_email(&denied_local_user_view, context.settings())?;
+      }
+      
+      LocalUser::delete(context.pool(), assessed_user_id).await?;
```

Comment:
You also have to delete from Person table.

Reply:
Not sure about this one. Deleting those from the tables does free up the IDs, but it also means we can't show them their deny reason, which we did before. I spose I'm good with it, if they didn't provide an email.

Reply:
Another thing about this: deleting their person will actually delete the entire registration application from the DB. Nothing will be known about their deny.